chore: added prettier config. removed doc comments. split and renamed…

breck7 opened the pull request:
… crawl method as prep for parallel refactor.

JustinBeckwith commented on package.json:
```diff
@@ -9,6 +9,14 @@
     "linkinator": "build/src/cli.js"
   },
+  "prettier": {
+    "useTabs": false,
```
Should be able to trim this to just be:
```
"singleQuote": true,
"trailingComma": "es5",
```

I think the rest are prettier defaults if I remember correctly

JustinBeckwith commented on src/index.ts:
```diff
@@ -8,6 +8,8 @@ import { getLinks } from './links';
 const finalhandler = require('finalhandler');
 const serveStatic = require('serve-static');
 
+declare type url = string;
```
At a glance, I'm not sure why we'd do that.  There is a `Url` type in node core thats _very_ different, and I'm worried that folks would confuse this type with that type.  Why doesn't `string` work?

breck7 replied:
I didn't think of the "Url" potential confusion. Good point. My thought is this is just a small step forward to more intelligent types. It just provides a hint that not all strings are valid, so if someone were to do something like crawl("http foo") in the future you potentially throw at dev/compile time instead of runtime. Very much a nit that I don't feel strongly for. Just something I do elsewhere (narrow all the general types) because at the least can help make things more self documenting
.

JustinBeckwith commented on src/index.ts:
```diff
@@ -33,39 +35,38 @@ export interface CrawlResult {
-/**
- * Instance class used to perform a crawl job.
- */
+const isHtml = (response: gaxios.GaxiosResponse) => {
```
Why move this to an arrow function?  Regular functions are honestly fine unless you're dealing with `this` pointer hell

breck7 replied:
No good reason. I just prefer arrows. A nit.

JustinBeckwith commented on src/index.ts:
```diff
@@ -33,39 +35,38 @@ export interface CrawlResult {
- * Instance class used to perform a crawl job.
- */
+const isHtml = (response: gaxios.GaxiosResponse) => {
+  const contentType = response.headers['content-type'] || '';
```
So I don't mind losing the jsdoc specific annotations, but I *do* like the comments explaining to me what was happening here :)

breck7 replied:
added back

JustinBeckwith commented on src/index.ts:
```diff
@@ -33,39 +35,38 @@ export interface CrawlResult {
+};
+
 export class LinkChecker extends EventEmitter {
-  /**
```
Ah! Let's keep the jsdoc headers, they weren't hurting anything

breck7 replied:
I've just learned never to trust jsdoc comments, unless they are statically checked for correctness. Otherwise they tend to become incorrect over time. Are these statically checked? I don't for a second disbelieve that your doc comments stay up to date, but most people (myself in particular) I would not trust that unless there's a static enforcement.

JustinBeckwith commented on src/index.ts:
```diff
@@ -76,14 +77,10 @@ export class LinkChecker extends EventEmitter {
-   * @returns Promise that resolves with the instance of the HTTP server
-   */
-  private startWebServer(root: string, port: number): Promise<http.Server> {
+  private _startStaticWebServer(
```
Yeah let's add the jsdoc back

breck7 replied:
Yeah I should have broken this up into 3 commits at least since they are orthogonal.

1. prettier.
2. crawl function split
3. js doc comments removal

Will do that at some point, as well as the interesting parallelization bit (if someone else doesn't beat me to it first). Great feedback, thanks!

breck7 replied:
added comments back

JustinBeckwith commented:
Thanks for starting the conversation with this PR!  Turns out - I was able to get it whipped up yesterday, and now the `--concurrency` flag is in the latest release :) Would love to hear what you think!

breck7 commented:
+100!!! So great!

I just ran this on about a dozen sites. Most of the sites completed in just a couple of seconds. Two of the bigger ones took a bit longer (mostly it seems just waiting for a few pages to timeout)

I already fixed a couple of stray broken links.

On one site which I haven't paid much attention to in a while it found `ERROR: Detected 40 broken links. Scanned 309 links in 79.046 seconds.` (Again, it seemed most of the links were crawled right away and a few took a while to error out making the final count longer than it felt). Most of them were because I switched the permalink naming scheme a while back and didn't pay attention to things, but now looking closely there was a drop in traffic I should have noticed. Thanks!

On one site it's checked about 13,000 links and found 130 broken links! That one still hasn't finished but the last 2 are reported as [0] so I'm guessing it's done other than a few things in the cue yet to timeout.

This is so great! Thanks!
